### PR TITLE
Feat/question service

### PR DESCRIPTION
**Pull request by tyouwei**
This PR aims to add create, update and delete operations to the `question-service`

### How to test
- Go to the postman docs
- Make sure that `POST localhost:4001/questions/` and `PUT localhost:4001/questions/:questionid` is able to take in a request body like
```
{
  "difficulty": [1..3],
  "description": string,
  "examples": string[],
  "constraints": string,
  "tags": string[],
  "title_slug": string,
  "title": string,
  "pictures": any[]
}
```
- Make sure that `DELETE localhost:4001/questions/:questionid` works and return the correct id

### CODE REVIEW

File: question-service/src/routes/deleteQuestionsController.ts
Comment (lsyurea) [inline question-service/src/routes/deleteQuestionsController.ts:1]: Hi You Wei, the function looks really good! Thanks for helping me create it. I believe the normal convention is to place all CRUD operations in one controller. Is it possible to help me shift all the delete question to create question to either the question or userquestion controller? Else we do not know which controller to use.

**Review by lsyurea — COMMENTED** [review]
Hello! We need to shift the functions based on standard CRUD API placement

File: question-service/src/routes/deleteQuestionsController.ts
Comment (lsyurea) [inline question-service/src/routes/deleteQuestionsController.ts:22]: The routes will be confusing as we do not know whether the delete is for user or for userquestion

File: question-service/src/app.ts
Comment (Kurtyjlee) [inline question-service/src/app.ts:7]: These imports should be removed if not cannot build

**Review by Kurtyjlee — COMMENTED** [review]
I see that `_question_id` is currently an int and is not unique. This might have some problems
1. When 2 objects have the same `_question_id`, the first one will be edited, which might not be the object you're intending to edit
2. We do not have an index for `_question_id` so the query will require a scan of the entire database, which is inefficient. Should use `_id` instead

Thanks for the PR tho, everything works well

File: question-service/src/app.ts
Comment (lsyurea) [reply question-service/src/app.ts]: Yups I was trying to remove them just now

File: question-service/src/routes/questionsController.ts
Comment (lsyurea) [inline question-service/src/routes/questionsController.ts:69]: This is not the convention. There should be a model used for question types

File: question-service/src/routes/questionsController.ts
Comment (lsyurea) [inline question-service/src/routes/questionsController.ts:83]: Question validation should be checked automatically.

**Review by Kurtyjlee — APPROVED** [review]
LGTM! After merge, can create a tag under `milestone-2`!

**Pull request merged**
